### PR TITLE
Done

### PR DESCRIPTION
**Pull request by ranelbe**
done

### CODE REVIEW

**Comment by ranelbe** [comment]
done

File: 5.1.no_vinaigrette.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+
+def generate_random_date(date1, date2):
+    """
+    Generate a random date between date1 and date2
+    :param date1: the first date
+    :param date2: the second date
+    :return: random date between date1 and date2
+    """
+    return date1 + (date2 - date1) * random.random()
```
Comment (excellenteam-scaleup) [inline 5.1.no_vinaigrette.py:39]: You should not aim to write everything in a single line, even if you can.
It's better to save date2-date1 to a variable to make it more explicit and clear. 
This link https://towardsdatascience.com/you-should-stop-craving-for-python-one-liners-5dbf8e0de23d discuss this topic

File: 5.1.thats_the_way.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
+"""
+Writer: Ranel Ben Simman Tov, 315967703
+"""
+
+import os
+
+
+def find_files_with_prefix(path, prefix='deep'):
```
Comment (excellenteam-scaleup) [inline 5.1.thats_the_way.py:8]: Please add a main function that uses this function, and proves your code is working.
add the files directory to the submission aswell

File: 5.2.slice_of_cake.py
Comment (excellenteam-scaleup) [inline 5.2.slice_of_cake.py:1]: I want a main function that test your code here aswell

File: 5.2_cup_of_join.py
Comment (excellenteam-scaleup) [inline 5.2_cup_of_join.py:1]: also here - main function

File: 5.4.connected_vessels.py
Comment (excellenteam-scaleup) [inline 5.4.connected_vessels.py:1]: Perfect!
Code is clean, well documented, and main function tests it

**Review by excellenteam-scaleup — CHANGES_REQUESTED** [review]

File: 5.1.no_vinaigrette.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+
+if __name__ == '__main__':
```
[the remark this answers is not in the document]
Comment (ranelbe) [reply 5.1.no_vinaigrette.py:42]: fixed

File: 5.1.no_vinaigrette.py (hunk 1 above)
Comment (ranelbe) [reply 5.1.no_vinaigrette.py:39]: fixed

File: 5.1.thats_the_way.py (hunk 2 above)
Comment (ranelbe) [reply 5.1.thats_the_way.py:8]: done

File: 5.2.slice_of_cake.py
Comment (ranelbe) [reply 5.2.slice_of_cake.py:1]: done

File: 5.2_cup_of_join.py
Comment (ranelbe) [reply 5.2_cup_of_join.py:1]: done

**Comment by ranelbe** [comment]
I'm sorry, I didn't know that adding Main to tests was mandatory.
I've made the necessary corrections.

**Review by excellenteam-scaleup — APPROVED** [review]

**Pull request merged**